aws/src/main/java/org/apache/iceberg/aws/sqs/SQSListener.java

```
@@ -63,5 +72,22 @@ public void notify(Object event) {
       LOG.error("Failed to add to queue", e);
     }
   }
+
+  @Override
+  public void initialize(String listenerName, Map<String, String> properties) {
+    AwsClientFactory factory = AwsClientFactories.from(properties);
+    this.sqs = factory.sqs();
+    if (listenerName != null) {
```

listener name must not be null, we should throw an error if it is

---

similar comment for SNS Listener, please fix and check null